unittests/snapshot_tests.cpp

```diff
@@ -692,20 +686,20 @@ static const char kv_snapshot_bios[] = R"=====(
 )=====";
 
 BOOST_AUTO_TEST_CASE_TEMPLATE(test_kv_snapshot, SNAPSHOT_SUITE, snapshot_suites) {
-   //for (bool rocks_save : { false, true }) {
-   //   for (bool rocks_load : { false, true }) {
-
-# warning TODO: Take out the return when snapshot is done with rewriting.
-   // snapshot handling is being rewritten.
-   // do not waste time to update soon-to-be-changed
-   // code to work. This is OK since we are in a
-   // development branch.
-   return;
-
-   for (bool rocks_save : { true }) {
-      for (bool rocks_load : { true }) {
+   for (bool rocks_save : { false, true }) {
+      for (bool rocks_load : { false, true }) {
          tester chain;
 
+         // set backing_store according to rocks_load value
+         chain.close(); // clean up chain so no dirty db error
+         auto cfg = chain.get_config();
+         if (rocks_load) {
+            cfg.backing_store = eosio::chain::backing_store_type::ROCKSDB;
+         } else {
+            cfg.backing_store = eosio::chain::backing_store_type::CHAINBASE;
+         }
+         chain.init(cfg); // enable new config
+
```

Comment:
I am thinking that there should be transactions/actions pushed that exercise kv_set, kv_erase, etc. so there is actual kv data in the snapshot, so we can see that the state of a nodeos started from snapshot matches a nodeos that actually applied the transactions.

Reply:
kv_snapshot_wast and kv_snapshot_bios achieve that. I verified via logging that kv_get, kv_set,... were called during the snapshot testing.

Reply:
Right.  kv_snapshot_wast stores a counter that is incremented in each loop iteration in kv.  This is based on the similar test for multi_index in `test_exhaustive_snapshot`.  These transactions were not changed by this PR, so if you're looking at this in github, you need to expand the hidden sections of this function to see them.